src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.ts

```diff
@@ -137,22 +187,95 @@ export class FlowsTableComponent implements OnInit, OnChanges {
                 noButtonText: "Cancel",
                 handler: (ok) => {
                     if (ok) {
-                        this.cancelFlowChange.emit(flowId);
+                        this.cancelFlowChange.emit({ flowId, datasetId });
                     }
                 },
             }),
         );
     }
 
-    public navigateToFlowDetaisView(flowId: string): void {
+    public navigateToFlowDetaisView(flow: FlowSummaryDataFragment, datasetId: string): void {
         this.navigationService.navigateToFlowDetails({
-            accountName: this.datasetBasics.owner.accountName,
-            datasetName: this.datasetBasics.name,
-            flowId,
+            accountName: this.accountView && this.accountName ? this.accountName : this.datasetBasics.owner.accountName,
+            datasetName: this.accountView ? this.datasetByIdForAccount(datasetId).name : this.datasetBasics.name,
+            flowId: flow.flowId,
         });
     }
 
     public dynamicImgSrc(status: FlowStatus): string {
         return DatasetFlowDetailsHelpers.dynamicImgSrc(status);
     }
+
+    public accountFormatter(x: Account | string): string {
+        return typeof x !== "string" ? x.accountName : x;
+    }
+
+    public searchByAccountName: OperatorFunction<string, readonly Account[]> = (text$: Observable<string>) =>
+        text$.pipe(
+            debounceTime(200),
+            distinctUntilChanged(),
+            tap(() => (this.searchingAccount = true)),
+            switchMap((term) =>
+                this.flowService.flowsInitiators(this.datasetBasics.id).pipe(
```

Comment:
If I'm not mistaken, you are re-querying flow initiators for every character user types. My heart is broken!